apps/mull-api/src/app/channel/channel.service.ts

```diff
@@ -1,26 +1,59 @@
 import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository } from 'typeorm';
-import { Channel } from '../entities';
-import { CreateChannelInput } from './inputs/channel.input';
+import { Channel, DirectMessageChannel, EventChannel } from '../entities';
+import { CreateDmChannelInput, CreateEventChannelInput } from './inputs/channel.input';
 @Injectable()
 export class ChannelService {
   constructor(
     @InjectRepository(Channel)
-    private channelRepository: Repository<Channel>
+    private channelRepository: Repository<Channel>,
+    @InjectRepository(EventChannel)
+    private eventChannelRepository: Repository<EventChannel>,
+    @InjectRepository(DirectMessageChannel)
+    private dmChannelRepository: Repository<DirectMessageChannel>
   ) {}
 
   getChannel(channelId: number): Promise<Channel> {
     return this.channelRepository.findOne(channelId);
   }
 
-  async createChannel(input: CreateChannelInput, eventId?: number): Promise<boolean> {
-    await this.channelRepository.save({ ...input, event: { id: eventId } });
+  getEventChannel(channelId: number): Promise<EventChannel> {
+    return this.eventChannelRepository.findOne(channelId);
+  }
+
+  getDmChannel(channelId: number): Promise<DirectMessageChannel> {
+    return this.dmChannelRepository.findOne(channelId, { relations: ['participants'] });
+  }
+
+  async createEventChannel(input: CreateEventChannelInput, eventId: number): Promise<boolean> {
+    await this.eventChannelRepository.save({ ...input, event: { id: eventId } });
     return true;
   }
 
+  async createDmChannel(input: CreateDmChannelInput): Promise<DirectMessageChannel> {
+    return this.dmChannelRepository.save(input);
+  }
+
   async deleteChannel(channelId: number): Promise<boolean> {
     await this.channelRepository.delete(channelId);
     return true;
   }
+
+  async findUsersInDmChannel(userId1: number, userId2: number): Promise<DirectMessageChannel> {
```

Comment:
The name is a little confusing. I thought this returns the users of a dm channel but it seems like it returns the channel given the users so it would be something like findDMChannelByUsersIds.

Reply:
Done